addons/web/static/src/views/fields/badge_selection/badge_selection_field.js

```diff
@@ -63,6 +65,38 @@ export class BadgeSelectionField extends Component {
         return JSON.stringify(value);
     }
 
+    get spanClass() {
+        if (this.props.colorField && this.props.record.data[this.props.name]) {
+            return `badge rounded-pill o_badge_color_${
+                this.props.record.data[this.props.colorField]
+            }`;
+        }
+        return "";
+    }
+
+    badgeSelectionClass(value) {
+        const badgeClass = [];
+        if (this.props.size === "sm") {
+            badgeClass.push("btn-sm");
+        }
+        if (this.props.size === "lg") {
+            badgeClass.push("btn-lg");
+        }
+        if (this.props.colorField) {
+            badgeClass.push("badge rounded-pill");
+            if (this.value === value) {
+                if (cookie.get("color_scheme") === "dark") {
+                    badgeClass.push("active o_badge_border border");
+                } else {
+                    badgeClass.push("active o_badge_border border");
+                }
+            }
+        } else if (this.value === value) {
+            badgeClass.push("active");
+        }
```

Comment:
`active` could be handled in `if (this.value === value)` 🙂 

```suggestion
        if (this.props.colorField) {
            badgeClass.push("badge rounded-pill");
            if (this.value === value) {
                badgeClass.push("o_badge_border border");
            }
        } 
        if (this.value === value) {
            badgeClass.push("active");
        }
```